### PR TITLE
Moved WaitNotifyService to impl

### PR DESCRIPTION
**Pull request by pveentjer**
The WaitNotifyService is an implementation detail and is not used
by the SPI user. So therefor it is moved to com.hazelcast.spi.waitnotify 

As part of the moving the InternalWaitNotifyService has been merged into the WaitNotifyService since it doesn't serve a lot of purpose to have 2 internal interfaces.

No Enterprise change needed.

### CODE REVIEW

**Comment by jerrinot** [comment]
:+1:

File: hazelcast/src/main/java/com/hazelcast/spi/BlockingOperation.java [hunk 1]
```diff
@@ -18,9 +18,7 @@
 
 
 /**
- * A interface that can be implemented to participate in the Wait/Notify System.
- *
- * See {@link  com.hazelcast.spi.WaitNotifyService}.
+ * A interface that can be implemented by an operation that can block. For example an ILock.acquire.
```
Comment (Donnerbart) [inline hazelcast/src/main/java/com/hazelcast/spi/BlockingOperation.java]: Nip picking, but it should be `An interface`.

**Comment by Donnerbart** [comment]
:+1:

**Comment by pveentjer** [comment]
Thanks. I'll fix the typo in a direct push.

**Comment by Donnerbart** [comment]
Already done :)

**Pull request merged**